### DISCUSSION TITLE
More examples in openxlsx2 style manual?

### DISCUSSION THREAD

**Discussion by mjona99** [JanMarvin/openxlsx2#371] [Q&A]
Hello,

First off, I want to thank you for the fantastic package!

I've struggled to implement the new style functions in `openxlsx2`. I think part of that is that with `openxlsx`, there were plenty of examples online (mainly stack overflow) that were able to guide my learning. 

I understand time + bandwidth is quite limited, but having a few more examples in the styling vignette would be immensely helpful in learning and understanding these new functions.

Many thanks again for the package!

**Comment by JanMarvin** [comment]
Hello @mjona99 ,

thanks for opening the discussion and thanks for the interest in `openxlsx2`. Short answer: we will make it! And you have all the time in the world to consider a transition, because `openxlsx` will not disappear. It will be there tomorrow and next week and in the next few months and it is tried and tested, we know all the strengths and weaknesses and nothing has to change at all. So a change to `openxlsx2` can happen, but it doesn't have to at all :)

The slightly longer answer is: we are still in an early stage of the package lifecycle. The intention of the only 10 days old CRAN release 0.3 was to let the world know that we exist and to get some first feedback (this is almost impossible with non-CRAN packages). Since then, we have already found and fixed a few bugs and received a first round of feedback. So version 0.3.1 will follow soon. And while we are, at least I personally, busting our asses to provide a package that meets our own high expectations, we know there are a few places we need to improve. A big part of that is the documentation, and we invite everyone to help us with that. Test our stuff, write bug reports or open pull requests.

Until then, if you have a small reproducible example with one of the built-in datasets (aka it runs in `reprex`), feel free to post it here and I will see about providing an `openxlsx2` solution. For me, it's easier when the questions are asked here. I also have a stackoverflow account, but there is a greater chance that I will miss the questions.

**Reply by mjona99** [reply]
Thank you for the quick response, @JanMarvin! 

Can't even imagine the undertaking it is to make a package, so I really do appreciate all your efforts! 

Here is a quick example I threw together using `mtcars`.

<img width="383" alt="image" src="https://user-images.githubusercontent.com/43893179/194988262-e4d4d52d-dfa9-402d-9358-0792d55c0734.png">

The main things that I've been struggling with:

- The fill working correctly 
- Different fonts (names, colors, and sizes) throughout the table 
- Accounting format (specifically the underline on the last line before 'Total')
- Grouping columns together

I've just begun my work with `openxlsx` and `openxlsx2` so I'm sure some of these might be trivial, but I appreciate any guidance you may be able to provide! Many thanks, again.

**Comment by JanMarvin** [comment]
Hi I was actually hoping you would add the regex example so that I can simply modify it. Still here is a quick example. There is a bug with the font color of the turquoise filled cell. Not sure what is picked up in this case.

```R
library(openxlsx2)

wb <- wb_workbook()$
  add_worksheet(gridLines = FALSE)$
  add_data(dims = "B2", x = "MTCARS Title")$
  add_data(x = head(mtcars), startCol = 2, startRow = 3)

# add font for title
wb$add_font(dims = "B2", bold = "1", size = "16")

# add font for head
wb$add_fill(dims = "B3:L3", color = wb_colour("turquoise"))
# # bug. somehow the wrong base cell is picked
# wb$add_font(dims = "B3:L3", color = wb_colour("white"))

wb$add_border(dims = "B9:L9", 
              bottom_color = wb_colour(hex = "FF000000"),
              bottom_border = "thin",
              left_border = "",
              right_border = "",
              top_border = "")

wb$open()
```
<img width="663" alt="Screenshot 2022-10-11 at 08 20 23" src="https://user-images.githubusercontent.com/1645626/195011512-18348812-5297-4e59-9063-33d689ea6d53.png">

**Reply by JanMarvin** [reply]
Follow up #372 a potential fix is in pull request #365
